site/en/blog/renderingng-architecture/index.md

```diff
@@ -0,0 +1,502 @@
+1. If not, the input event is sent back to the browser process.
+1. The browser process converts it to a scroll gesture by combining it with other recent events.
+1. The scroll gesture is sent once again to baz.com's render process compositor thread,
+1. The scroll is applied there, and the compositor thread for the bar.com
+render process ticks an animation in its compositor event loop.
+This then mutates scroll offset in the property trees and re-runs the compositor lifecycle.
+It also tells the main thread to fire a scroll event (not depicted here).
+1. A compositor frame is sent to Viz.
+1. Viz aggregates the compositor frames for the foo.com render process,
+the bar.com render process, and the browser UI.
+1. Viz schedules a draw.
+1. Viz draws the aggregated compositor frame to the screen.
+
+{% Aside %}
+Note that each input event after the first can skip steps three and four,
+because the scroll has already begun and at that point can observe it,
+but no longer interrupt.
+{% endAside %}
+
+To route a click event on a hyperlink in iframe #two on tab one:
+
+{% Img src="image/ZDZVuXt6QqfXtxkpXcPGfnygYjd2/GnXCsuXJkNegGDMQAhnx.jpeg",
+alt="", width="800", height="488" %}
+
+1. An input event (mouse, touch or keyboard) comes to the browser process.
+It performs an approximate hit test
+to determine that the bar.com render process should receive the click, and sends it there.
+1. The compositor thread for bar.com routes the click event to the main thread
+for bar.com and schedules a rendering event loop task to process it.
+1. The input event processor for bar.com's main thread hit tests to determine which
+DOM element in the iframe was clicked, and fires a click event for scripts to observe.
+Hearing no preventDefault, it navigates to the hyperlink.
+1. Upon load, the new state is rendered, with steps similar to the "render changed DOM" example above.
+(These subsequent changes are not depicted here.)
+
+## Conclusion
+
+Phew, that was a lot of detail.
+As you can see, rendering in Chromium is quite complicated.
+It can take a lot of time to remember and internalize all the pieces,
+so don't be worried if it seems overwhelming.
+
+The most important takeaway is that there is a conceptually simple rendering pipeline,
+which through careful modularization and attention to detail,
+has been split into a number of self-contained components.
+These components have then been split across parallel processes and threads to maximize
+[scalable performance](/blog/renderingng/#scalable-performance)
+and [extensibility](/blog/renderingng/#extensibility-the-right-tools-for-the-job) opportunities.
+
+Each of those components plays a critical role in enabling
+all of the performance and features modern web apps need.
+Next we'll publish deep-dives into each of them,
+and the important role they play.
+
+But before that, I'll also explain how the key data structures mentioned in this post
```

Comment:
```suggestion
But before that, next up I'll explain how the key data structures mentioned in this post
```

Reply:
I wondered about "But before that, the next post will explain ..." Before that, next up is a bit of an odd construction.

Reply:
Maybe:

"Soon we'll publish deep-dives..."

And leave this sentence alone.